src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -218,14 +220,24 @@ def create_connectedk8s(cmd, client, resource_group_name, cluster_name, https_pr
     config_dp_endpoint = get_config_dp_endpoint(cmd, location)
 
     # Retrieving Helm chart OCI Artifact location
-    registry_path = os.getenv('HELMREGISTRY') if os.getenv('HELMREGISTRY') else utils.get_helm_registry(cmd, config_dp_endpoint, dp_endpoint_dogfood, release_train_dogfood)
+    if container_registry_repository:
+        if disable_auto_upgrade is False:
```

Comment:
Instead of throwing error here, should we give a y/n prompt saying: 'using private container registry will disable arc-agent auto update. Do you want to continue?' and then internally set auto_upgrade false if user says 'y'?
For first time onboarding, we will see this failure a lot of times as folks might not be aware to explicitly disable auto update.
But if we are documenting this somewhere, then it should be fine. In fact, we should add this point in the description of this parameter in params.py.

Reply:
ack